sentry/src/main/java/io/sentry/Scopes.java

```diff
@@ -421,6 +421,7 @@ public void close(final boolean isRestarting) {
         }
 
         // TODO: should we end session before closing client?
+        // TODO [HSM] should we go through all clients (global, isolation, current) and close them?
```

Comment:
I think it would make sense to close `isolation` and `current` at least, not sure about `global`.

Before HSM we only closed the top-most `client` on the stack

```
        // Close the top-most client
        final StackItem item = stack.peek();
        // TODO: should we end session before closing client?
        item.getClient().close(isRestarting);
```

Reply:
I'd say global should also be closed since this is likely just being forwarded from `Sentry.close()`.